pkg/kvstore/allocator/doublewrite/backend.go

```diff
@@ -175,6 +175,55 @@ func (d *doubleWriteBackend) UpdateKey(ctx context.Context, id idpool.ID, key al
 	return crdErr
 }
 
+func (d *doubleWriteBackend) UpdateMasterKey(ctx context.Context, id idpool.ID, key allocator.AllocatorKey, reliablyMissing bool) error {
```

Comment:
I've mostly just copied the logic from `UpdateKey` to keep the existing behavior.